internal/models/articles/articles.go

```diff
@@ -100,11 +100,16 @@ func scanArticleFromFullRow(row *sql.Row) (*Article, error) {
 		return nil, err
 	}
 
-	// Parse the image URL.
-	tempImageURL, err := url.Parse(imageURL)
-	article.ImgURL = *tempImageURL
-	if err == sql.ErrNoRows {
-		return nil, err
+	// Convert the image URL from DB to a URL type.
+	if imageURL.Valid {
+		tempURL, err := url.Parse(imageURL.String)
+		if err != nil {
+			return nil, err
+		} else {
+			article.ImgURL = *tempURL
+		}
+	} else {
+		article.ImgURL = url.URL{}
```

Comment:
Code is repeated multiple times, is it possible to create a function?

Reply:
Fixed in [0a07123](https://github.com/Molnes/Nyhetsjeger/pull/52/commits/0a07123ede6ca8dcbc1e58d154b4d574e1ed3365)